application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -280,6 +326,66 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContent, @Nullable String previousContent) {
+
+        if (List
+            .of(Subcommand.CREATE, Subcommand.CREATE_WITH_MESSAGE, Subcommand.EDIT,
+                    Subcommand.EDIT_WITH_MESSAGE)
+            .contains(subcommand) && newContent == null) {
+            logger.debug("newContent is null even though the subcommand should supply a value.");
+            return;
+        }
+
+        if (List.of(Subcommand.EDIT, Subcommand.EDIT_WITH_MESSAGE, Subcommand.DELETE)
+            .contains(subcommand) && previousContent == null) {
+            logger
+                .debug("previousContent is null even though the subcommand should supply a value.");
```

Comment:
why is this just `debug`? sounds like a `warning`, no?

why do you silently support this without crashing? please stick to **fail-fast**. if there is a bug in the system, throw an exception. `throw new IllegalArgumentException(...)` or dont even support `@Nullable` on the argument.

Reply:
changed it, see 8b6807158246ecd79c1ecbd7ec43720149c10b58